File: docs/js_vs_py/01_function_parameters.js
const Math2 = require('./libs/math'); // Overwrites the standard Math module
console.log(Math.random())


const addNameToList = (nameToAdd, list=[]) => {
    list.push(nameToAdd);
    return list;
}

console.log(addNameToList('First Name'))
console.log(addNameToList('Second Name'))

/**
 * Function parameters are evaluated at every single function invocation
 */

File: docs/js_vs_py/01_function_parameters.py


def addNameToList(name, list = []):
    list.append(name)
    return list

print(addNameToList('First Name'))
print(addNameToList('Second Name'))
print(addNameToList('Fourth Name',  ['Fifth name']))

print("** ")
print(" Watch out with mutable function arguments!")
print("** ")


if (False):
    def addNameToListInPython(name, list = None):
        if list is None:
            list = []

        list.append(name)
        return list

    print(addNameToListInPython('First Name'))
    print(addNameToListInPython('Second Name'))
    print(addNameToListInPython('Fourth Name',  ['Fifth name']))

"""
Function parameters are read and interpreted only ONCE when parsing the file
"""

